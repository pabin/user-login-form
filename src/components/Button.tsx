import React from 'react';
import styles from './styles/button.module.css';

type ButtonProps = {
  title: string;
  disabled: boolean;
  onClick: () => void;
}

const Button  = ({ title, onClick, disabled }: ButtonProps) => (
  <button className={styles.button} onClick={onClick} disabled={disabled ? true : false}>
    { title }
  </button>
)

export default Button;