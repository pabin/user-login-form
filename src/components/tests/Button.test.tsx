import React from 'react';
import { render, fireEvent, cleanup, screen } from '@testing-library/react'
import Button from '../Button';

describe('Button', () => {
  const title = 'Next';
  const mockedOnClick = jest.fn();
  afterEach(cleanup)

  test('renders correctly when button is disabled', async () => {
    const wrapper = render(<Button title={title} onClick={mockedOnClick} disabled/>);

    expect(wrapper.baseElement).toMatchSnapshot();
    expect(wrapper.getByText(title)).toBeInTheDocument()
  });

  test('renders correctly when button is enabled', () => {
    const wrapper = render(<Button title={title} onClick={mockedOnClick} disabled={false}/>);

    expect(wrapper.baseElement).toMatchSnapshot();
    expect(wrapper.getByText(title)).toBeInTheDocument()
  });

  test('call onClick when pressed and is enabled', async () => {
    render(<Button title={title} onClick={mockedOnClick} disabled={false}/>);

    const button = screen.getByRole('button', { name: title});
    fireEvent.click(button);

    expect(mockedOnClick).toHaveBeenCalled();
  });

  test('should not call onClick when pressed and is disabled', async () => {
    render(<Button title={title} onClick={mockedOnClick} disabled />);

    const button = screen.getByRole('button', { name: title });
    fireEvent.click(button);

    expect(mockedOnClick).not.toHaveBeenCalled();
  });
});