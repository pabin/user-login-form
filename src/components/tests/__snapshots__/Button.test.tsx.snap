// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly when button is disabled 1`] = `
<body>
  <div>
    <button
      class="button"
      disabled=""
    >
      Next
    </button>
  </div>
</body>
`;

exports[`Button renders correctly when button is enabled 1`] = `
<body>
  <div>
    <button
      class="button"
    >
      Next
    </button>
  </div>
</body>
`;
